### ISSUE TITLE
Chained goeswith

### ISSUE THREAD

**Issue by martinpopel** [UniversalDependencies/docs#414]
How to annotate more than two tokens which should be written together, but the original text (incorrectly) separates them?
My suggestion (confirmed by the current practice in UD_Croatian) is they should form a right-to-left or left-to-right chain (not siblings). For example
```
1 never ... 2 goeswith
2 the   ... 3 goeswith
3 less  ... 0 root
```

This would have the nice (testable by [svalidation](http://universaldependencies.org/svalidation.html)) property that `goeswith` always links two neighboring tokens.
Using this test, I was able to find several annotation errors, e.g.
```
─┮ has VERB conj
 │ ╭─╼ no X goeswith
 ╰─┾ where NOUN goeswith
   │ ╭─╼ to PART mark
   ╰─┶ go VERB acl
```

Also, I think we should test that the first of the two nodes linked by `goeswith` does not have `SpaceAfter=No`.
Again, this allows us the svalidation to find possibly incorrectly used `goeswith`.

So I am asking just for confirmation my understanding is correct, so I can publish the tests (and maybe clarify the [goeswith specification](http://universaldependencies.org/u/dep/goeswith.html) a bit).

**Comment by jnivre** [comment]
It seems more consistent with the general UD scheme to use a left-headed structure where all non-first elements attach to the first (as for "fixed" and "flat"):

1 never ... 0 root
2 the   ... 1 goeswith
3 less  ... 1 goeswith

Would this make things worse for validation purposes?

**Comment by martinpopel** [comment]
@jnivre: there are two questions: 1. head-final vs. head-initial vs. mixed-head and 2. chain vs. flat analysis.
1. The current [goeswith specification](http://universaldependencies.org/u/dep/goeswith.html) says "The head is in some sense the “main” part, often the second part." So either we need to change this specification, or we need to accept that goeswith can be sometimes head-final and sometimes head-initial. Personally, I have a slight preference for consistency and changing the specification. I have no preference wrt head-final vs. head-initial.
2. In UDv1.4 Croatian and Romanian uses goeswith chains consistently. Russian uses both chain and flat analysis, but in both cases there are many annotation errors (it seems they use this deprel instead of `SpaceAfter=No`). Validating chains is slightly simpler, but I can easily validate the flat structure analysis in Udapi as well (I am not sure about the SETS query language).

I can convert any treebank to the head-initial flat analysis automatically, if needed. So we should not be too concerned with the legacy data and backward compatibility.
We just need to agree what are the guidelines for goeswith.

**Comment by dan-zeman** [comment]
I think I would prefer to change the specification and treat `goeswith` as a technical relation, with a prescribed structure as in `fixed` (and `flat`). It is not always clear what is the “main” part, and there may be no good candidate (consider an extreme example where every letter is separated: _w i t h i n_).

**Comment by jnivre** [comment]
Sounds good to me.

**Comment by martinpopel** [comment]
Closed via #417
